### ISSUE TITLE
np.einsum will not accept NoneType for 'out' parameter

### ISSUE THREAD

**Issue by whagar** [numpy/numpy#15776]
np.einsum raises an error when NoneType ('None') is passed to the 'out' keyword argument. This behavior is not in line with the expected behavior for the out keyword argument of ufunc methods.

### Reproducing code example:

```python
import numpy as np
np.einsum('i,j',[1],[2],out=None)
```

### Error message:
```
Traceback (most recent call last):

  File "<ipython-input-2-32138e47a353>", line 1, in <module>
    np.einsum('i,j',[1],[2],out=None)

  File "<__array_function__ internals>", line 6, in einsum

  File "einsumfunc.py", line 1356, in einsum
    return c_einsum(*operands, **kwargs)

TypeError: keyword parameter out must be an array for einsum
```

### Numpy/Python version information:

1.18.1 3.7.5 (default, Oct 31 2019, 15:18:51) [MSC v.1916 64 bit (AMD64)]

**Comment by eric-wieser** [comment]
Should be a fairly straightforward fix, possibly one that can be made purely in python. Thanks for the report.

**Comment by sgasse** [comment]
Is this still an issue? I can reproduce the bug in an older version like 1.17.2, but not on the latest master:
```
>>> import numpy as np
>>> np.__version__
'1.19.0.dev0+0b9b176'
>>> np.einsum('i,j',[1],[2],out=None)
array([[2]])
```

**Comment by WarrenWeckesser** [comment]
@eric-wieser, you already fixed this, in https://github.com/numpy/numpy/pull/15256! 😄

Should we add a unit test to ensure it stays fixed?

**Comment by sgasse** [comment]
Ah OK :)
@WarranWeckesser: If we could use a unit test, could I give it a try adding it? I'd like to try out the workflow here.

**Comment by rossbar** [comment]
This should be fully closed by the combination of #15256 & #15797
